### PR TITLE
Miscellaneous API

### PR DESCRIPTION
**Pull request by Thundermaker300**
* Player.KickPower
* Player.VoiceColor
* Player.VoiceChannel
* Ragdoll.IsConsumed
* Scp049Role.IsCallActive
* (method) Scp049Role.CanResurrect (two overloads)
* Scp0492Role.ResurrectNumber
* Scp0492Role.SimulatedStare
* Scp0492Role.BloodlustActive
* Scp0492Role.IsConsuming
* Scp0492Role.RagdollConsuming (todo: come up with a better name for this)
* (method) Scp049Role.IsInConsumeRange (two overloads)
* Scp939Role.MimicryCooldown
* (method) Map.PlaceAmnesticCloud
* Renamed: Ragdoll.AllowRecall -> Ragdoll.IsExpired (since this property does not check extra conditions for dead zombies)

### CODE REVIEW

File: Exiled.API/Features/Player.cs [hunk 1]
```diff
@@ -708,6 +713,26 @@ public bool IsIntercomMuted
...
+
+        /// <summary>
+        /// Gets or sets the player's voice channel.
+        /// </summary>
+        public VoiceChatChannel VoiceChannel
+        {
+            get => VoiceModule?.CurrentChannel ?? VoiceChatChannel.None;
+            set
+            {
+                if (VoiceModule is null)
```
Comment (iopietro) [inline Exiled.API/Features/Player.cs:729]:
```suggestion
                if (VoiceModule == null)
```

File: Exiled.API/Features/Roles/Scp0492Role.cs [hunk 2]
```diff
@@ -40,12 +50,60 @@ public float AttackDamage
...
+        /// Gets or sets a value indicating the amount of time to simulate SCP-049-2's Bloodlust ability.
+        /// </summary>
+        public float SimulatedStare
+        {
+            get => SubroutineModule.TryGetSubroutine(out ZombieBloodlustAbility ability) ? ability.SimulatedStare : 0;
+            set
+            {
+                if (!SubroutineModule.TryGetSubroutine(out ZombieBloodlustAbility ability))
+                    return;
+                ability.SimulatedStare = value;
```
Comment (iopietro) [inline Exiled.API/Features/Roles/Scp0492Role.cs:63]:
```suggestion
                if (!SubroutineModule.TryGetSubroutine(out ZombieBloodlustAbility ability))
                    return;

                ability.SimulatedStare = value;
```

File: Exiled.API/Features/Roles/Scp0492Role.cs [hunk 3]
```diff
@@ -40,12 +50,60 @@ public float AttackDamage
...
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether or not SCP-049-2 is currently pursuing a target (Bloodlust ability).
+        /// </summary>
+        public bool BloodlustActive
+        {
+            get => SubroutineModule.TryGetSubroutine(out ZombieBloodlustAbility ability) ? ability.LookingAtTarget : false;
+        }
```
Comment (iopietro) [inline Exiled.API/Features/Roles/Scp0492Role.cs:73]:
```suggestion
        public bool BloodlustActive => SubroutineModule.TryGetSubroutine(out ZombieBloodlustAbility ability) ? ability.LookingAtTarget : false;
```

**Review by iopietro — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Player.cs [hunk 4]
```diff
@@ -708,6 +713,26 @@ public bool IsIntercomMuted
...
+        /// Gets the player's voice color.
+        /// </summary>
+        public Color VoiceColor => ReferenceHub.serverRoles.GetVoiceColor();
+
+        /// <summary>
+        /// Gets or sets the player's voice channel.
+        /// </summary>
+        public VoiceChatChannel VoiceChannel
+        {
+            get => VoiceModule?.CurrentChannel ?? VoiceChatChannel.None;
```
[the remark this answers is not in the document]
Comment (Thundermaker300) [reply Exiled.API/Features/Player.cs:726]: damn

**Pull request merged**